tikv/raftstore/fsm_apply.go

```diff
@@ -446,23 +615,121 @@ func (d *applyDelegate) updateMetrics(aCtx *applyContext) {
 }
 
 func (d *applyDelegate) writeApplyState(wb *WriteBatch) {
-	// TODO: stub
+	wb.Set(ApplyStateKey(d.region.Id), d.applyState.Marshal())
 }
 
 func (d *applyDelegate) handleRaftEntryNormal(aCtx *applyContext, entry *eraftpb.Entry) applyResult {
-	return applyResult{} // TODO: stub
+	index := entry.Index
+	term := entry.Term
+	if len(entry.Data) > 0 {
+		cmd := new(raft_cmdpb.RaftCmdRequest)
+		err := cmd.Unmarshal(entry.Data)
```

Comment:
Should check the term and index after unmarshal from bytes.

Reply:
but tikv doesn't check term and index.

Reply:
in parse_data_at.